src/Workspaces/Core/Portable/EmbeddedLanguages/VirtualChars/VirtualChar.cs

```diff
@@ -61,6 +79,8 @@ public override int GetHashCode()
         public static bool operator !=(VirtualChar char1, VirtualChar char2)
             => !(char1 == char2);
 
-        public static implicit operator char(VirtualChar vc) => vc.Char;
+        public static implicit operator Rune(VirtualChar vc) => vc.Rune;
+        public static implicit operator int(VirtualChar vc) => vc.Rune.Value;
+        public static implicit operator string(VirtualChar vc) => vc.ToString();
```

Comment:
❗️ All of these need to be removed

* Conversion to `Rune` is lossy
* Conversion to `int` allows this type to be used in nonsensical arithmetic expressions
* Conversion to `string` forces an allocation

```suggestion
```

Reply:
> Conversion to Rune is lossy

I don't see any issue with that.  It's not like the 'Rune' portion loses precision.  It's more a form of simulating struct inheritance.  i.e. you can think of this as `struct VirtualChar : Rune`.  It adds info about the span, but it certainly could be used where a Rune is wanted/needed without any problems.

I'm fine removing the other two.